packages/docs/docs/guides/5-request-client/2-erc20-payment-detection.ts

```diff
@@ -0,0 +1,211 @@
+ * Proxy contract payment network parameters are the same as the address based payment network: the payment and eventually the refund address.
+ */
+
+const proxyContractPaymentNetwork: RequestNetwork.Types.Payment.IPaymentNetworkCreateParameters = {
+  id: RequestNetwork.Types.Payment.PAYMENT_NETWORK_ID.ERC20_PROXY_CONTRACT,
+  parameters: {
+    // eslint-disable-next-line spellcheck/spell-checker
+    paymentAddress: '0x92FC764853A9A0287b7587E59aDa47165b3B2675',
+  },
+};
+
+const proxyContractCreateParams = {
+  proxyContractPaymentNetwork,
+  requestInfo,
+  signer: payeeIdentity,
+};
+
+// Finally create the request and print its id
+(async () => {
+  const request = await requestNetwork.createRequest(proxyContractCreateParams);
+
+  console.log(`Request created with erc20 proxy contract payment network: ${request.requestId}`);
+
+  // The payer can add a refund address. For example when accepting the request
+  await request.accept(payerIdentity, {
+    refundAddress: '0xfb7914D846ab64A44c4e65dd38F4D004d29F911F',
+  });
+})();
+
+// eslint-disable-next-line spellcheck/spell-checker
+/**
+ * ### Paying an erc20 proxy contract request manually
+ * 
+ * Payments for requests using ERC20 proxy contract payment network are documented in an Ethereum smart contract: *ERC20Proxy*. This smart contract is available on Ethereum [mainnet](https://etherscan.io/address/0x5f821c20947ff9be22e823edc5b3c709b33121b3) and Rinkeby [testnet](https://rinkeby.etherscan.io/address/0x162edb802fae75b9ee4288345735008ba51a4ec9).
+ * To perform a payment, the payer has first to allow the smart contract to transfer funds from his address. You must call `approve(spender, amount)` method of the specific ERC20 token smart contract. `spender` is the address of the proxy smart contract (`0x5f821c20947ff9be22e823edc5b3c709b33121b3` on mainnet, `0x162edb802fae75b9ee4288345735008ba51a4ec9` on Rinkeby), `amount` is the maximum amount you want to allow the proxy contract to transfer funds.
+ * Since the payer could eventually pay other requests,  amount can be `2**32 - 1` (maximum value of a uint32 in solidity language) so that the payer only has to call `approve` once.
+ * Then, payment can be performed by calling `transferFromWithReference(tokenAddress, to, amount, paymentReference)` method. `tokenAddress` is the address of the ERC20 token, to is the Ethereum address of the payee, `amount` is the amount of the payment. The user has to provide a payment reference to link the payments to the request. The payment reference is the last 8 bytes of a salted hash of the requestId: `last8Bytes(hash(lowercase(requestId + salt + address)))`.
+ * Any payments documented in the ERC20 proxy contract with the correct reference is considered as a payment for the request. The payments made with this method can be retrieved in the `TransferWithReference` event logs of the proxy smart contract.
+ */
```

Comment:
open question: is the above really necessary? we could link to the docs to know more, but a beginner doesn't need to know exactly how this works, only which methods to call in order to pay the request (and yes, a tiny explanation on the approval)

Reply:
I agree, it can be tricky sometimes. This specific section is small enough to be skipped easily by beginners.

Reply:
+1 remove this, we should have this documented in a more technical doc somewhere.